spec.html

```diff
@@ -36563,27 +36563,137 @@
+          1. Let _env_ be a new Lexical Environment.
+          1. Let _envRec_ be a new block Environment Record containing no bindings.
+          1. Let _fibRec_ be a new declarative Environment Record containing no bindings.
+          1. Set _envRec_.[[FunctionsInBlock]] to _fibRec_.
+          1. Set _env_'s EnvironmentRecord to be _envRec_.
+          1. Set the outer lexical environment reference of _env_ to _E_.
+          1. Return _env_.
+        </emu-alg>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-block-static-semantics-early-errors">
+        <h1>Changes to Block Early Errors</h1>
+        <p>In Static Semantics: Early Errors of <emu-grammar>Block : `{` StatementList `}`</emu-grammar> (<emu-xref href="#sec-block-static-semantics-early-errors"></emu-xref>) the following replaces the first list item:</p>
+        <ul>
+          <li>
+            It is a Syntax Error if the LexicallyDeclaredNames of StatementList contains any duplicate entries, unless the source code matching this production is not strict mode code and the duplicate entries are only bound by FunctionDeclarations.
+          </li>
+        </ul>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-block-runtime-semantics-evaluation">
+        <h1>Changes to Block Runtime Evaluation</h1>
+        <p>During runtime evaluation of <emu-grammar>Block : `{` StatementList `}`</emu-grammar> (<emu-xref href="#sec-block-runtime-semantics-evaluation"></emu-xref>) the following step is performed in place of step 2:</p>
+        <emu-alg>
+          1. Let _blockEnv_ be NewBlockEnvironment(_oldEnv_).
+        </emu-alg>
+        <p>During runtime evaluation of <emu-grammar>SwitchStatement : `switch` `(` Expression `)` CaseBlock</emu-grammar> (<emu-xref href="#sec-switch-statement-runtime-semantics-evaluation"></emu-xref>) the following step is performed in place of step 4:</p>
+        <emu-alg>
+          1. Let _blockEnv_ be NewBlockEnvironment(_oldEnv_).
+        </emu-alg>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-blockdeclarationinstantiation">
+        <h1>Changes to BlockDeclarationInstantiation</h1>
+        <p>During BlockDeclarationInstantiation (<emu-xref href="#sec-blockdeclarationinstantiation"></emu-xref>) the following step is performed in place of step 2:</p>
+        <emu-alg>
+          1. Assert: _envRec_ is a block Environment Record.
+        </emu-alg>
+        <p>The following steps are performed in place of step 4:</p>
+        <emu-alg>
+          1. Let _functionNames_ be an empty List.
+          1. Let _functionsToInitialize_ be an empty List.
+          1. For each _d_ in _declarations_, in reverse list order do
+            1. For each element _dn_ of the BoundNames of _d_ do
+              1. If IsConstantDeclaration of _d_ is *true*, then
+                1. Perform ! _envRec_.CreateImmutableBinding(_dn_, *true*).
+              1. Else,
+                1. Perform ! _envRec_.CreateMutableBinding(_dn_, *false*).
```

Comment:
It's necessary to check for duplicate names before calling CreateMutableBinding (assertion in [step 2 of CreateMutableBinding](http://tc39.github.io/ecma262/#sec-declarative-environment-records-createmutablebinding-n-d)). 

This version of the algorithm avoids this issue:
```md
1. For each element _d_ in _declarations_ do
  1. If _d_ is not a |FunctionDeclaration|, then
    1. For each element _dn_ of the BoundNames of _d_ do
      1. If IsConstantDeclaration of _d_ is *true*, then
        1. Perform ! _envRec_.CreateImmutableBinding(_dn_, *true*).
      1. Else,
        1. Perform ! _envRec_.CreateMutableBinding(_dn_, *false*).
    1. If _d_ is a |GeneratorDeclaration|, then
      1. Let _fn_ be the sole element of the BoundNames of _d_.
      1. Let _fo_ be the result of performing InstantiateFunctionObject for _d_ with argument _env_.
      1. Perform _envRec_.InitializeBinding(_fn_, _fo_).
1. Let _functionNames_ be an empty List.
1. Let _fibRec_ be the value of _envRec_.[[FunctionsInBlock]].
1. For each element _d_ in _declarations_, in reverse list order do
  1. If _d_ is a |FunctionDeclaration|, then
    1. Let _fn_ be the sole element of the BoundNames of _d_.
    1. If _fn_ is not an element of _functionNames_, then
      1. NOTE If there are multiple |FunctionDeclaration|s for the same name, the last declaration is used. There are only multiple declarations if _code_ is not contained in strict mode code.
      1. Append _fn_ to _functionNames_.
      1. Perform ! _envRec_.CreateMutableBinding(_fn_, *false*).
      1. Perform ! _fibRec_.CreateMutableBinding(_fn_, *false*).
      1. Let _fo_ be the result of performing InstantiateFunctionObject for _d_ with argument _env_.
      1. Perform _envRec_.InitializeBinding(_fn_, _fo_).
      1. Perform _fibRec_.InitializeBinding(_fn_, _fo_).
```